### PR TITLE
Fix build error due to upstream change

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

Fix build error due to upstream changes 
https://github.com/opensearch-project/common-utils/pull/655
https://github.com/opensearch-project/anomaly-detection/pull/1210

 
### Issues Resolved

https://github.com/opensearch-project/skills/issues/311
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Hailong-am** [comment]
integTest failed due to, that will need to a fix at AD side

> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.integTest.SearchAnomalyDetectorsToolIT.testSearchAnomalyDetectorsToolInFlowAgent_detectorStateParams" -Dtests.seed=676D44C4FE298B5C -Dtests.security.manager=false -Dtests.locale=ar-SA -Dtests.timezone=Europe/Sofia -Druntime.java=17
    {"error":{"reason":"System Error","details":"class org.opensearch.timeseries.transport.GetConfigRequest cannot be cast to class org.opensearch.timeseries.transport.GetConfigRequest (org.opensearch.timeseries.transport.GetConfigRequest is in unnamed module of loader java.net.FactoryURLClassLoader @752973de; org.opensearch.timeseries.transport.GetConfigRequest is in unnamed module of loader java.net.FactoryURLClassLoader @43c39321)","type":"ClassCastException"},"status":500}

File: build.gradle [hunk 1]
```diff
@@ -126,7 +126,7 @@ dependencies {
     // Plugin dependencies
     compileOnly group: 'org.opensearch', name:'opensearch-ml-client', version: "${opensearch_build}"
     implementation fileTree(dir: jsJarDirectory, include: ["opensearch-job-scheduler-${opensearch_build}.jar"])
-    implementation fileTree(dir: adJarDirectory, include: ["opensearch-time-series-analytics-${opensearch_build}.jar"])
+    implementation fileTree(dir: adJarDirectory, include: ["opensearch-anomaly-detection-${opensearch_build}.jar"])
```
Comment (zane-neo) [inline build.gradle:129]: AD team plans to change the jar name from `opensearch-anomaly-detection` to `opensearch-time-series-analytics`, but main and 2.x and release branch are not consistent, during 2.14 release, this inconsistency caused skills failed to build. @kaituo can you confirm which name will be using in main and 2.x in 2.15 release?

Comment (Hailong-am) [reply build.gradle:129]: Based on latest main, @kaituo changed back to `opensearch-anomaly-detection` in [#1210](https://github.com/opensearch-project/anomaly-detection/pull/1210/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7L206)

**Comment by Hailong-am** [comment]
@zane-neo please help to re-run the CI, class cast exception fixed in AD and i could pass the build in my local.

**Review by zane-neo — APPROVED** [review]

**Pull request merged**